cli/azd/pkg/apphost/generate.go

```diff
@@ -107,11 +110,15 @@ func Dockerfiles(manifest *Manifest) map[string]genDockerfile {
 	return res
 }
 
+type AppHostManager struct {
```

Comment:
I sort of dislike adding the manager here if we don't need to - in favor of just passing this (perhaps as part of some larger options bag that could also hold `autoConfigureDataProtection`) to the top level functions, which would remain "static" (i.e. no receiver).  I think it makes it easier to reason about and flexible.  Did it look really awful when we didn't have this stuff hanging off the manager?

Reply:
For what it's worth, I have found the decreased reusability aspect of objects carrying a bunch of state around to be a valid concern.

I was reminded of the banana/gorilla/jungle quote recently:
“… Because the problem with object-oriented languages is they’ve got all this implicit environment that they carry around with them. You wanted a banana but what you got was a gorilla holding the banana and the entire jungle. “ —Joe Armstrong, creator of Erlang progamming language

Reply:
you never know when having a gorilla by your side could be useful, @weikanglim  =P  jk.  good quote.